src/MessagePack/MessagePackWriter.cs

```diff
@@ -923,5 +839,79 @@ public void WriteExtensionFormat(ExtensionResult extensionData)
             WriteExtensionFormatHeader(extensionData.Header);
             WriteRaw(extensionData.Data);
         }
+
+        /// <summary>
+        /// Writes a 16-bit integer in big endian format.
+        /// </summary>
+        /// <param name="value">The integer.</param>
+        public void WriteBigEndian(ushort value)
```

Comment:
Force inlining?

Reply:
When do you recommend I go to the trouble of doing so? Can I assume that at reasonable times the JIT will inline methods on its own without me adding the AggressiveInlining attribute?

Reply:
Yes it’s supposed to based on some heuristic. You could look at the generated asm

Reply:
I believe PerfView also shows the JIT events that describe inlining. I'll check it out holistically for these two types. I do agree though: this method is a great candidate for inlining.
But no one calls *this* method (yet). At the moment, I'm supposing that it's faster, even with the wrapper `BufferWriter` type, for the caller to acquire just one span and write to it multiple times rather than calling GetSpan and Write several times. And all methods that want to write out big endian have a 1-2 byte header to emit first, so all of these call the overloads that accept a Span instead of this method, which calls GetSpan, write, and Advance.
Does that seem worthwhile? Otherwise I can call this method from all those places and simplify all the callers.

Reply:
I think it’s fine to do it later just something to look out for. There’s also a trick you can do which is writing into the span backwards (or highest index first). Bounds checks get eliminated for subsequent writes to the span.